src/node_modules/app/machines/builder/form.options.js

```diff
@@ -66,9 +64,9 @@ function computeLists (ctx) {
 		effaced: [aggDicts, datasetDicts, fieldDicts]
 		.some(isMissing('types', type))
 	})));
-	ctx.datasetOptions.set(DATASETS.map((dset, index) => ({
+	ctx.datasetOptions.set(DATASETS.map(dset => ({
 		text: dset.id,
-		value: index,
+		value: dset.id,
 		disabled: [typeDicts, fieldDicts, aggDicts]
 		.some(isMissing('datasets', dset.id))
```

Comment:
I'd indent these like:

```suggestion
		disabled:
			[typeDicts, fieldDicts, aggDicts]
			.some(isMissing('datasets', dset.id))
```